packages/hasura-storage-js/src/machines/file-upload.ts

```diff
@@ -96,76 +96,41 @@ export const createFileUploadMachine = () =>
-          }
           const file = (event.file || context.file)!
-          headers['x-nhost-file-name'] = event.name || file.name
           const data = new FormData()
           data.append('file', file)
-          if (event.adminSecret) {
-            headers['x-hasura-admin-secret'] = event.adminSecret
-          }
-          if (event.accessToken) {
-            headers['Authorization'] = `Bearer ${event.accessToken}`
-          }
+
           let currentLoaded = 0
-          const controller = new AbortController()
-          axios
-            .post<{
-              bucketId: string
-              createdAt: string
-              etag: string
-              id: string
-              isUploaded: true
-              mimeType: string
-              name: string
-              size: number
-              updatedAt: string
-              uploadedByUserId: string
-            }>(event.url + '/files', data, {
-              headers,
-              signal: controller.signal,
-              onUploadProgress: (event: AxiosProgressEvent) => {
-                const loaded = event.total
-                  ? Math.round((event.loaded * file.size!) / event.total)
-                  : 0
-                const additions = loaded - currentLoaded
-                currentLoaded = loaded
-                callback({
-                  type: 'UPLOAD_PROGRESS',
-                  progress: event.total ? Math.round((loaded * 100) / event.total) : 0,
-                  loaded,
-                  additions
-                })
-              }
-            })
-            .then(({ data: { id, bucketId } }) => {
-              callback({ type: 'UPLOAD_DONE', id, bucketId })
-            })
-            .catch(({ response, message }: AxiosError<{ error?: { message: string } }>) => {
+
+          fetchUpload(event.url, data, {
+            fileId: event.id || context.id,
+            bucketId: event.bucketId || context.bucketId,
+            accessToken: event.accessToken,
+            adminSecret: event.adminSecret,
+            name: event.name || file.name,
+            onUploadProgress: (event) => {
+              const loaded = event.total ? Math.round((event.loaded * file.size!) / event.total) : 0
+              const additions = loaded - currentLoaded
+              currentLoaded = loaded
               callback({
-                type: 'UPLOAD_ERROR',
-                error: {
-                  status: response?.status ?? 0,
-                  message: response?.data?.error?.message || message,
-                  // TODO errors from hasura-storage are not codified
-                  error: response?.data?.error?.message || message
-                }
+                type: 'UPLOAD_PROGRESS',
+                progress: event.total ? Math.round((loaded * 100) / event.total) : 0,
+                loaded,
+                additions
               })
-            })
+            }
+          }).then(({ fileMetadata, error }) => {
```

Comment:
Can we use `async-await` instead of `then`?

Reply:
we can't here, as the xstate service is running on callback mode. If it finds an `async` in the method, it will then run on promise mode, which we want to avoid